lib/iris/fileformats/grib/_load_convert.py

```diff
@@ -158,11 +158,21 @@ def _unscale(v, f):
 
     if isinstance(value, Iterable) or isinstance(factor, Iterable):
         def _masker(item):
-            result = ma.masked_equal(item, _MDI)
+            # This is a small work around for an edge case, which is not
+            # evident in any of our sample GRIB2 messages, where an array
+            # of header elements contains missing values.
+            # iris.fileformats.grib.message returns these as None, but they
+            # are wanted as a numerical masked array, so a temporary mdi
+            # value is used, selected from a legacy implementation of iris,
+            # to construct the masked array. The valure is transient, only in
+            # scope for this function.
+            numerical_mdi = 2 ** 32 - 1
+            item = [numerical_mdi if i is None else i for i in item]
```

Comment:
I believe not.  If this were implemented, then a mixed type underlying array would be created, with floats and None, so NumPy would have to make an object array.  This would be a behaviour change, which I don't think is desired.